cineast-api/src/main/java/org/vitrivr/cineast/api/util/QueryUtil.java

```diff
@@ -261,10 +261,14 @@ private static void retrieveAndWeight(ContinuousRetrievalLogic continuousRetriev
     List<String> _return = new ArrayList<>();
     DBSelector selector = Config.sharedConfig().getDatabase().getSelectorSupplier().get();
     selector.open(SegmentTags.SEGMENT_TAGS_TABLE_NAME);
-    List<Map<String, PrimitiveTypeProvider>> rows = selector.getRows(GENERIC_ID_COLUMN_QUALIFIER, ids);
-
-    rows.forEach(row -> _return.add(row.get(SegmentTags.TAG_ID_QUALIFIER).getString()));
-    return _return;
+    try {
+      List<Map<String, PrimitiveTypeProvider>> rows = selector.getRows(GENERIC_ID_COLUMN_QUALIFIER, ids);
+      rows.forEach(row -> _return.add(row.get(SegmentTags.TAG_ID_QUALIFIER).getString()));
+      return _return;
+    } catch (Exception e) {
+      LOGGER.error("Exception while looking up tags", e);
+      return new ArrayList<>();
```

Comment:
Since you are already allocating `_return`, you can return it after the try/catch block in either case